### ISSUE TITLE
Rename package `akka.typed` to `akka.typed.actor` as other things are also typed

### ISSUE THREAD

**Issue by ktoso** [akka/akka#23675]
Since we're going all in and all modules are getting typed API, we should rename `akka.typed.Actor` etc to `akka.typed.actor.Actor`, as repetetive as it sounds it is true that that "typed" is "typed actors", and "typed" will eventually be all of Akka, so this rename should be done before it's too late.

**Comment by patriknw** [comment]
This is a duplicate of https://github.com/akka/akka/issues/23632

**Comment by ktoso** [comment]
Ah, missed that one, thanks
